caffe2/operators/top_k.cc

```diff
@@ -26,147 +33,184 @@ template <typename T>
 struct ValueCmp {
   bool operator()(
       const std::pair<T, TIndex>& lhs,
-      const std::pair<T, TIndex>& rhs) {
-    return (
-        lhs.first > rhs.first ||
-        (lhs.first == rhs.first && lhs.second < rhs.second));
+      const std::pair<T, TIndex>& rhs) const {
+    return lhs.first > rhs.first ||
+        (lhs.first == rhs.first && lhs.second < rhs.second);
   }
 };
 
-// Define these two names to allow lookup into the 2d tensors like
-// mytensor(i, j)
 template <typename T>
-using EigenMatrixMapRowMajor = Eigen::Map<
-    Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>>;
+void GetTopK(
```

Comment:
Nit: briefly add some simple comments about the core algorithm/formula. It will be very helpful to understand the code.